client/src/main/java/com/aliyun/emr/rss/client/compress/RssZstdCompressor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aliyun.emr.rss.client.compress;
+
+import java.util.zip.CRC32;
+import java.util.zip.Checksum;
+
+import com.github.luben.zstd.Zstd;
+
+public class RssZstdCompressor extends RssZstdTrait implements Compressor {
+  private final int compressionLevel;
+  private final Checksum checksum;
+  private byte[] compressedBuffer;
+  private int compressedTotalSize;
+
+  public RssZstdCompressor() {
+    this(256 * 1024);
+  }
+
+  public RssZstdCompressor(int blockSize) {
+    // Using 1 by default
+    compressionLevel = 1;
+    // Using CRC32 which is suitable for error detect rather than
+    checksum = new CRC32();
+    initCompressBuffer(blockSize);
+  }
+
+  @Override
+  public void initCompressBuffer(int maxDestLength) {
+    int compressedBlockSize = HEADER_LENGTH + maxDestLength;
+    compressedBuffer = new byte[compressedBlockSize];
```

Comment:
In our internal testing on Spark vanilla shuffle, w/ using `RecyclingBufferPool`, zstd has significant GC overhead compared to lz4

Reply:
Ditto

Reply:
we can keep as-is and observe the actual GC performance, then to see if we need to use buffer pool, because the compressor's buffer array may not be frequently initialized under its usage scenario.

Reply:
thx for explanation, make sense to me